### ISSUE TITLE
new type inference: incorrect warning about boxed enum

### ISSUE THREAD

**Issue by nicks** [google/closure-compiler#1447]
Code:

``` js
/** @enum {number} */
var XType = {ONE: 1}

/** @constructor */ function X() { 
  /** @type {XType} */ this.type = XType.ONE; 
}

/**
 * @param {XType|X} x
 * @return {number}
 */
function f(x) {
  return typeof x == 'number' ? x : x.type
}
```

Expected result: no warnings

Actual result:

```
input0:11: WARNING - Property type may not be present on x of type Number|X.
  return typeof x == 'number' ? x : x.type
                                    ^
```

Old type inference handles this correctly. I suspect that it's autoboxing the enum in a weird place?

http://closure-compiler-debugger.appspot.com/#input0%3D%252F**%2520%2540enum%2520%257Bnumber%257D%2520*%252F%250Avar%2520XType%2520%253D%2520%257BONE%253A%25201%257D%250A%250A%252F**%2520%2540constructor%2520*%252F%2520function%2520X()%2520%257B%2520%252F**%2520%2540type%2520%257BXType%257D%2520*%252F%2520this.type%2520%253D%2520XType.ONE%253B%2520%257D%250A%250A%252F**%250A%2520*%2520%2540param%2520%257BXType%257CX%257D%2520x%250A%2520*%2520%2540return%2520%257Bnumber%257D%250A%2520*%252F%250Afunction%2520f(x)%2520%257B%250A%2520%2520return%2520typeof%2520x%2520%253D%253D%2520'number'%2520%253F%2520x%2520%253A%2520x.type%250A%257D%26input1%26conformanceConfig%26externs%26refasterjs-template%26includeDefaultExterns%3D1%26CHECK_SYMBOLS%3D1%26CHECK_TYPES%3D1%26CHECK_TYPES_NEW_INFERENCE%3D1%26CLOSURE_PASS%3D1%26LANG_IN_IS_ES6%3D1%26MISSING_PROPERTIES%3D1%26PRESERVE_TYPE_ANNOTATIONS%3D1%26PRETTY_PRINT%3D1%26TRANSPILE%3D1
